### DISCUSSION TITLE
Send form after 'event.preventDefault()`

### DISCUSSION THREAD

**Discussion by alexandremjacques** [unpoly/unpoly#691] [Q&A]
Hi,

I have an use case where one form field is set to disabled after changing an option. Since I still need it's value on submit, I would like to set the disabled property to false before submitting the form.

I'm listening to the `up:form:submit` and I was able to `event.preventDefault()` and enable the field. But if I try to `up.submit(event.form)` the form I guess I end up in a loop (since I'm submitting the form the event is firing up again).

If I skip the `event.preventDefault()`, I think the form is submitted before I have the chance to change the disabled field.

Is there a way to make Unpoly resume the form submission only after making this change?

TIA

**Comment by adam12** [comment]
I might catch the `up:form:submit` event and then mutate the event object with the parameters I want.

```javascript
up.on('up:form:submit', event => {
  event.form['the_disabled_field'].disabled = false

  // Start with an empty params
  event.params.clear()
  
  // Re-parse form with disabled field enabled
  event.params.addAll(up.Params.fromForm(event.form))
})
```


Here's a Glitch that seems to work, demonstrating what I mean: https://glitch.com/edit/#!/season-nebulous-carp?path=scripts.js%3A11%3A46

**Comment by triskweline** [comment]
HTML also has a [`[readonly]`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly) attribute. This makes a field immutable, but still includes it in the form parameters.

**Reply by alexandremjacques** [reply]
I've tried that but, since the field is a `<select>` element, it still functions (user can change the selection in the UI). Not a very good UX. 😞 

I haven't tried @adam12 suggestion yet. I'll come back here with the results.

Thanks!
